src/main/java/com/linecorp/armeria/client/routing/ServerGroupRegistry.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.routing;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * A memory server group register center.
+ */
+public class ServerGroupRegistry {
+    private static final Map<String, NodeSelector> serverGroups = new ConcurrentHashMap<>();
+
+    /**
+     * Register the server group.if the groupName has already registered,throw
+     * {@link ServerGroupRegisteredException}.
+     */
+    public static boolean register(String groupName, ServerGroup serverGroup,
+                                   NodeSelectStrategy nodeSelectStrategy) {
+        requireNonNull(groupName, "groupName");
+        requireNonNull(serverGroup, "serverGroup");
+        requireNonNull(nodeSelectStrategy, "nodeSelectStrategy");
+
+        NodeSelector existSelector = serverGroups.putIfAbsent(groupName,
+                nodeSelectStrategy.newSelector(serverGroup));
```

Comment:
Could you check if `serverGroups` already contains the group with the given `groupName` before creating a new selector, so that we do not create a new selector unnecessarily?